scene/gui/split_container.cpp

```diff
@@ -368,6 +364,15 @@ bool SplitContainer::is_vertical() const {
 	return vertical;
 }
 
+void SplitContainer::set_dragging_enabled(bool p_enabled) {
+	dragging_enabled = p_enabled;
```

Comment:
If dragging is disabled while it is being dragged, it doesn't end properly and can cause issues next time it is enabled.

Also the mouse cursor isn't updated, you might need to call Viewport's `update_mouse_cursor_state` or something.

Reply:
I had not anticipated disabling while dragging, so let me have a look at handling that correctly...

Reply:
I tested the new push disabling while dragging, and the mouse cursor updates correctly.

Reply:
When I test it the mouse cursor doesn't update until it is next moved

Reply:
I'm trying variations on the below and the cursor won't update until movement. Is there another class I can examine that has the behavior of updating the cursor while hovering and stationary after a state changes? What about using an `InputEventMouse` to trigger the update?
```
dragging_area_control->get_viewport()->update_mouse_cursor_state();
```

Reply:
Hmm, that should work. `update_mouse_cursor_state` is what `set_default_cursor_shape` uses to updated it immediately.
I guess if it isn't working then it can be looked into later.

Reply:
Yes, I could not get that to work. Check out this new push and see if it does the trick. I'm also using an `InputEventMouseMotion` to cause a jiggle. I think it updates correctly when dragging and also when hovering.

Reply:
It works when disabling, but not when enabling. I think there is an issue with the method in Viewport, it can be fixed later.

For now we can just use:

```c
if (get_viewport()){
    get_viewport()->update_mouse_cursor_state();
}
```

Reply:
Okay, I put in the viewport call that we can revisit later. Should be good for now.